x-pack/plugins/event_log/server/es/names.ts

```diff
@@ -22,10 +22,13 @@ export interface EsNames {
 export function getEsNames(baseName: string): EsNames {
   const eventLogName = `${baseName}${EVENT_LOG_NAME_SUFFIX}`;
   const eventLogNameWithVersion = `${eventLogName}${EVENT_LOG_VERSION_SUFFIX}`;
+  const eventLogPolicyName = `${
+    baseName.startsWith('.') ? baseName.substring(1) : baseName
+  }${EVENT_LOG_NAME_SUFFIX}`;
   return {
     base: baseName,
     alias: eventLogNameWithVersion,
-    ilmPolicy: `${eventLogName}-policy`,
+    ilmPolicy: `${eventLogPolicyName}-policy`,
```

Comment:
nit: can we update `x-pack/plugins/event_log/server/es/names.mock.ts` to reflect the change (remove `.` prefix).